platform/darwin/src/MGLMultiPoint.mm

```diff
@@ -93,6 +98,45 @@ - (void)getCoordinates:(CLLocationCoordinate2D *)coords range:(NSRange)range
     }
 }
 
+- (void)setCoordinates:(CLLocationCoordinate2D *)coords range:(NSRange)range
+{
+    if ([self isMemberOfClass:[MGLMultiPoint class]])
+    {
+        [[NSException exceptionWithName:@"MGLAbstractClassException"
+                                 reason:@"MGLMultiPoint is an abstract class"
+                               userInfo:nil] raise];
+    }
+
+    NSAssert(range.length > 0, @"Empty coordinate range");
+
+    [self willChangeValueForKey:@"coordinates"];
+    if (range.location >= _count)
```

Comment:
There are a few different cases we need to consider. The current code doesn’t address all of them. I think this code would be simpler with the following algorithm:

1. The new range is entirely contained within `_coordinates`, in which case `NSMaxRange(range) <= _count`. Patch up `_coordinates` using `memcpy()`. (Note that the `num` parameter to `memcpy()` is the number of bytes.)
2. The new range begins within `_coordinates` but extends beyond it, in which case `NSMaxRange(range) > _count`. Allocate a new array on the heap, `memcpy()` `_coordinates` to the beginning of it, then set `_coordinates` to that new array.
3. The new range doesn’t overlap `_coordinates` at all, in which case `range.location > _count`. Raise an exception.

Since (1) will reuse the memory that `_coordinates` already points to, we should avoid freeing that memory unless we know we’re in (2) and we’re already done with the old array.